src/execution_algorithms/sgd_training_algorithm.cpp

```diff
@@ -149,7 +149,11 @@ void SGDTrainingAlgorithm::train(SGDExecutionContext& c,
 
     // Finalize epoch
     c.inc_epoch();
-    model.reconcile_weight_values();
+
+    // FIXME (tbennun 10/09/23): This is weight averaging and should not be done
+    // unless requested. Commented out for now to test for any regressions.
```

Comment:
What you're saying is, that this is an invariant that shouldn't be enforced anyway in model parallel mode, and will do nothing in the data parallel mode (since weights are STARxSTAR). In the standard synchronous SGD (and all of its variants), the updates should yield the same weights. If reproducibility is a concern, allreduce would introduce floating point inconsistencies just as much as the individual gradient updates would. Other deep learning frameworks do not introduce this additional grid-wide communication call per weight. There should be no reason to keep this reconciliation; we commented it out to see if anything starts to fail and nothing did.